apps/src/music/player/MusicPlayer.ts

```diff
@@ -345,6 +393,11 @@ export default class MusicPlayer {
   }
 
   getLengthForId(id: string): number | null {
+    // Hack to allow patterns to render in the same way as samples. Will undo this when tweaking pattern timeline UI.
```

Comment:
I know I added these, but looking at this again, I think the simplest way to get around this would just be to pass the entire `eventData` object into the `TimelineElement` component. That way, the component can directly check if type === 'pattern' and then not have to look up its ID using getLengthForId or getTypeForId. We could also then remove the `id` field from `PatternEvent`.

Edit: we could also then probably remove `'pattern'` from the `SoundType` interface in MusicLibrary since we won't need to call getTypeForId

Reply:
Interesting.  I think I've done that now.